### PR TITLE
Drag and drop added for images in the sidebar

### LINKED ISSUE #644: Order images in the sidebar using drag and drop
**Description**
The user must be able to move the images in the sidebar and position them.
Peek 2019-05-28 11-11

### PR DESCRIPTION
**Pull request by divyabaid16**
Fixes #644  (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

[//]: # (To mark checkbox write 'x' within the square brackets)

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [ ] ask `@publiclab/reviewers` for help, in a comment below

GIF For the same:
Peek 2019-05-28 11-11

Thanks!

### CODE REVIEW

**Comment by jywarren** [comment]
Hmm, I'm not sure about that error, but maybe googling for the exact phrase
with the word "field" especially could help -- it doesn't look like the
typical "that column doesn't exist" type error.

I was wondering if we could think through what would happen if a warpable
belonged to multiple maps. How could we track position? Would we put the
position field in the join table, something like image_to_map.position, so
each map could have a different ordering? Or what other ideas do you have?

Thanks!!!

**Comment by divyabaid16** [comment]
@jywarren 

Here, after drag and drop of the table rows, I tried to get the order in which they are present now, 
it can be seen in
```
Processing by ImagesController#sort as */*
  Parameters: {"warpable"=>["100", "101", "99"]}

```

And, after getting the parameters, I updated the ```position```, i.e a new colunm and displayed the table in sorted order of position according the the way, the params are sorted. 

And regarding, warpable belonging to multiple map, I think while doing this, id of the particular map was recorded and changes was made only to the images in that map.

Thanks!

**Comment by divyabaid16** [comment]
My error for [this comment](https://github.com/publiclab/mapknitter/pull/645#issuecomment-496516304) is solved. 

I had cleaned all the tables and migrated it once again. 

Thanks for your help @jywarren !

**Comment by jywarren** [comment]
I guess this is a separate discussion then, about
images-belonging-to-many-maps, but if we did do this, i think it may be
worth putting the position in the join table, or each time reordering
happens, it could rearrange other maps too!

See here for reference -- does this make sense?
https://guides.rubyonrails.org/association_basics.html#the-has-and-belongs-to-many-association

Thanks!

**Comment by divyabaid16** [comment]
I got your point @jywarren , I will try it now.
Thanks!

**Comment by divyabaid16** [comment]
Hi @jywarren @gauravano @sashadev-sky 
I'm still not able to reorder the images after the page is refreshed even though I had ordered the images in the database based on the position.
Can you please help me out with this?

Thanks!

File: app/views/images/_index.html.erb [hunk 1]
```diff
@@ -3,7 +3,7 @@
   <p id="no-images">No images uploaded. </p>
   <%end%>
 <table class="warpables-all table table-striped">
-  <tbody>
+  <tbody data-url = "<%= sort_images_path %>">
     <% @map.warpables.each do |warpable| %>
```
Comment (sashadev-sky) [inline app/views/images/_index.html.erb]:
```suggestion
    <% ary = @map.warpables.sort { |a, b| a.position <=> b.position } %>
    <% ary.each do |warpable| %>
```

**Comment by sashadev-sky** [comment]
My commit suggestion also still works if you delete the update to the `feeds_controller`:  
```
@warpable = Warpable.sort_by('position')
```

and in the images_controller can delete:
```
 @warpable = Warpable.order('position')    
    respond_to do |format|
      format.json { head :no_content }
      format.js   { render :layout => false }
```

Although i'm wondering about the `respond_to` block, is this just a RESTful way to handle not responding any content so should be kept? and why do we now say we respond to `format.js`?

**Comment by divyabaid16** [comment]
Oh, I guess I had added that lines to feed_controller by mistake. 
It has nothing to do with that.

Thanks!

**Comment by divyabaid16** [comment]
Thank you @sashadev-sky for your changes suggested :)

I'm attaching the working GIF
Peek 2019-05-30 21-20

@jywarren @gauravano can you please review this now?

Thanks!

**Comment by divyabaid16** [comment]
The test is failing because of too many lines in image_controller.

**Comment by sashadev-sky** [comment]
@divyabaid16 try rebasing and testing it - because its throwing a server error for me now!

**Comment by divyabaid16** [comment]
I have created a new pull request #670 
Thanks!

**Pull request closed without merging**